src/redux/rockets/rocketsSlice.js

```diff
@@ -1,4 +1,23 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const formattedBooks = (data) => data.map((rocket) => ({
+  id: rocket.id,
+  rocketname: rocket.rocket_name,
+  type: rocket.rocket_type,
+  image: rocket.flickr_images[0],
+}));
+
+export const getRocketData = createAsyncThunk('rockets/getRocketData', async () => {
+  try {
+    const rocketsDataLink = 'https://api.spacexdata.com/v3/rockets';
+    const response = await fetch(rocketsDataLink);
+    const data = await response.json();
```

Comment:
As an optional suggestion, you can use axios to save like 2 lines of code, and handle the json part.